library/src/test/java/com/novoda/downloadmanager/DownloadManagerTest.java

```diff
@@ -0,0 +1,281 @@
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().withStatus(DownloadBatchStatus.Status.DOWNLOADING).build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        InOrder inOrder = inOrder(downloadBatch);
+        inOrder.verify(downloadBatch).status();
+        inOrder.verifyNoMoreInteractions();
+    }
+
+    @Test
+    public void removesBatchFromInternalList_whenResuming() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        assertThat(downloadBatches).doesNotContainEntry(DOWNLOAD_BATCH_ID, downloadBatch);
+    }
+
+    @Test
+    public void resumesBatch() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        verify(downloadBatch).resume();
+    }
+
+    @Test
+    public void triggersDownload_whenResumingBatch() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        verify(downloadManagerDownloader).download(downloadBatch, downloadBatches);
+    }
+
+    @Test
+    public void doesNotDelete_whenBatchIdIsUnknown() {
+        downloadManager.delete(new LiteDownloadBatchId("unknown"));
+
+        verifyZeroInteractions(downloadBatch, additionalDownloadBatch);
+    }
+
+    @Test
+    public void removesBatchFromInternalList_whenDeleting() {
+        downloadManager.delete(DOWNLOAD_BATCH_ID);
+
+        assertThat(downloadBatches).doesNotContainEntry(DOWNLOAD_BATCH_ID, downloadBatch);
+    }
+
+    @Test
+    public void deletesBatch() {
+        downloadManager.delete(DOWNLOAD_BATCH_ID);
+
+        verify(downloadBatch).delete();
+    }
+
+    @Test
+    public void addsCallbackToInternalList() {
+        DownloadBatchCallback additionalDownloadBatchCallback = mock(DownloadBatchCallback.class);
+
+        downloadManager.addDownloadBatchCallback(additionalDownloadBatchCallback);
+
+        assertThat(downloadBatchCallbacks).contains(additionalDownloadBatchCallback);
+    }
+
+    @Test
+    public void removesCallbackFromInternalList() {
+        downloadManager.removeDownloadBatchCallback(downloadBatchCallback);
+
+        assertThat(downloadBatchCallbacks).doesNotContain(downloadBatchCallback);
+    }
+
+    @Test(timeout = 500)
+    public void waitsForServiceToExist_whenGettingAllBatchStatuses() throws InterruptedException {
+        final ArgumentCaptor<Runnable> argumentCaptor = ArgumentCaptor.forClass(Runnable.class);
+        willAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                argumentCaptor.getValue().run();
+                return null;
+            }
+        }).given(executorService).submit(argumentCaptor.capture());
```

Comment:
do we need a captor? The parameter is passed into he answer

Reply:
I never realised this before ❤️